src/pages/instances/InstanceDetailHeader.tsx

```diff
@@ -13,61 +14,56 @@ interface Props {
 }
 
 const InstanceDetailHeader: FC<Props> = ({ name, instance, project }) => {
-  const [isRename, setRename] = useState(false);
-  const canRename = instance?.status === "Stopped";
+  const navigate = useNavigate();
+  const notify = useNotify();
+  const [isSubmitting, setSubmitting] = useState(false);
+
+  const handleRename = (newName: string): Promise<void> => {
+    return new Promise((resolve, reject) => {
+      if (name === newName) {
+        return resolve();
+      }
+      setSubmitting(true);
+      renameInstance(name, newName, project)
+        .then(() => {
+          navigate(
+            `/ui/${project}/instances/detail/${newName}`,
+            notify.queue(notify.success("Instance renamed."))
+          );
+          resolve();
+        })
+        .catch((e) => {
+          notify.failure("Renaming failed", e);
+          reject();
+        })
+        .finally(() => setSubmitting(false));
+    });
+  };
 
   return (
-    <div className="p-panel__header">
-      <h1 className="u-off-screen">{name}</h1>
-      {instance ? (
-        <div className="p-panel__title">
-          <nav
-            key="breadcrumbs"
-            className="p-breadcrumbs"
-            aria-label="Breadcrumbs"
-          >
-            <ol className="p-breadcrumbs__items">
-              <li className="p-breadcrumbs__item">
-                <Link to={`/ui/${project}/instances`}>Instances</Link>
-              </li>
-              {isRename ? (
-                <li className="p-breadcrumbs__item instance-rename">
-                  <InstanceRename
-                    instance={instance}
-                    project={project}
-                    closeForm={() => setRename(false)}
-                  />
-                </li>
-              ) : (
-                <li
-                  className="p-breadcrumbs__item instance-name u-truncate"
-                  onClick={() => canRename && setRename(true)}
-                  title={name}
-                >
-                  <Tooltip
-                    message={!canRename && "Stop the instance to rename"}
-                    position="btm-left"
-                  >
-                    {name}
-                  </Tooltip>
-                </li>
-              )}
-            </ol>
-          </nav>
-          {!isRename && (
-            <div>
-              <i className="status u-text--muted">{instance.status}</i>
-              <InstanceStateActions key="state" instance={instance} />
-            </div>
-          )}
-        </div>
-      ) : (
-        <h4 className="p-panel__title">{name}</h4>
-      )}
-      <div className="p-panel__controls">
-        {instance && !isRename && <DeleteInstanceBtn instance={instance} />}
-      </div>
-    </div>
+    <RenameHeader
+      name={name}
+      parentItem={<Link to={`/ui/${project}/instances`}>Instances</Link>}
+      renameDisabledReason={
+        !instance || instance.status !== "Stopped"
```

Comment:
Shouldn't this be:
```suggestion
        instance?.status !== "Stopped"
```
?

Reply:
Why was this first committed and then reverted? :thinking:

Reply:
The change should be represented, I force pushed it with the other changes.

Reply:
Right, now it is included.